src/graph/Graph.java

```diff
@@ -0,0 +1,155 @@
+package graph;
+
+import java.util.*;
+
+public class Graph {
+
+  private Map<Integer, Vertex> vertices;
+  private String name;
+
+  public Graph() {
+    this("unnamed");
+  }
+
+  public Graph(String name) {
+    this.vertices = new HashMap<>();
+    this.name = name;
+  }
+
+  public String getName() {
+    return name;
+  }
+
+  public void setName(String name) {
+    this.name = name;
+  }
+
+  public Vertex get_node(int id) {
+    return this.vertices.get(id);
+  }
+
+  public Set<Integer> node_ids() {
+    return vertices.keySet();
+  }
+
+  public Collection<Vertex> nodes() {
+    return vertices.values();
+  }
+
+  public Vertex add_node(Vertex v) {
+    this.vertices.put(v.getId(), v);
+    return v;
+  }
+
+  public Vertex add_node(int id) {
+    Vertex node = new Vertex(id);
+    return this.add_node(node);
+  }
+
+  public Vertex add_node() {
+    if (this.is_empty()) {
+      return this.add_node(0);
+    }
+    int max_id = Collections.max(this.vertices.keySet());
```

Comment:
As you are already using a hash map, I do not recommend this interface. Assume id is not continuous.